examples/webvr_controllers.html

```diff
@@ -0,0 +1,187 @@
+
+				//
+
+				controls = new THREE.VRControls( camera );
+				controls.standing = true;
+
+				effect = new THREE.VREffect( renderer );
+
+				if ( WEBVR.isAvailable() === true ) {
+
+					document.body.appendChild( WEBVR.getButton( effect ) );
+
+				}
+
+				//
+
+				window.addEventListener( 'resize', onWindowResize, false );
+
+				// controller stuff
+				window.addEventListener( 'vr controller connected', onVRControllerConnected );
+
+
+				// dat gui
+				dat.GUIVR.enableMouse( camera );
+
+      			const gui = dat.GUIVR.create( 'Settings' );
+      			gui.position.x = 0.15;
+      			gui.position.z = -1.00;
+      			gui.position.y = 1.5;
+      			gui.rotation.y = -Math.PI / 4;
+      			scene.add( gui );
+
+				gui.add( torus.position, 'x', -1, 1 ).step(0.001).name('Position X');
+				gui.add( torus.position, 'y', -1, 1 ).step(0.001).name('Position Y');
+				gui.add( torus.rotation, 'y', -Math.PI, Math.PI).step(0.001).name('Rotation').listen();
+			}
+
+			function onVRControllerConnected(event) {
+				const controller = event.detail;
+				scene.add( controller );
+
+				// for 6dof
+				controller.standingMatrix = window.controls.getStandingMatrix()
+
+				// for 3dof
+				controller.head = window.camera
+
+
+				// create a mesh
+				var meshColorOff = 0xff4040;
+				var meshColorOn = 0xffff00;
+				var controllerMaterial = new THREE.MeshStandardMaterial( { color: meshColorOff, shading: THREE.FlatShading } );
+				var controllerMesh = new THREE.Mesh(
+					new THREE.CylinderGeometry(0.005, 0.05, 0.1, 6),
+					controllerMaterial
+				);
+				controllerMesh.rotation.x = -Math.PI/2;
+
+				var handleMesh = new THREE.Mesh( new THREE.BoxGeometry(0.03,0.1,0.03), controllerMaterial);
+				handleMesh.position.y = -0.05;
+				controllerMesh.add(handleMesh)
+
+				controller.userData.mesh = controllerMesh; // so we can change the color later
+				controller.add(controllerMesh);
+
+
+				// enable gui
+				const guiInputHelper = dat.GUIVR.addInputObject( controller );
+			    scene.add( guiInputHelper );
+
+				controller.addEventListener( 'primary press began', function(ev){
+					const mesh = ev.target.userData.mesh;
+					mesh.material.color.setHex(meshColorOn);
+					guiInputHelper.pressed(true);
+				})
+				controller.addEventListener( 'primary press ended', function(ev){
+					const mesh = ev.target.userData.mesh;
+					mesh.material.color.setHex(meshColorOff);
+					guiInputHelper.pressed(false);
+
+				})
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				effect.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			function animate() {
+
+				THREE.VRController.update();
+
+				effect.requestAnimationFrame( animate );
+				render();
+
+			}
+
+			function render() {
+
+				torus.rotation.y +=  0.002;
+				if (torus.rotation.y > Math.PI) torus.rotation.y =  torus.rotation.y - (Math.PI * 2);
+
+				controls.update();
```

Comment:
Is your comment about `THREE.VRControls`? Or about this proposed `THREE.VRController`? (Both follow the convention of requiring the developer to include a call to `.update()` in their animation loop.) I chose to follow the existing `THREE.VRControls` convention because I think this is the cleanest way to do it—rather than having independent `rAF` loops; makes the sequence of updates more predictable.